### PR TITLE
Trend analysis changes drastically or moderately

### LINKED ISSUE #64: Incorrect "Steep drop-off"

### PR DESCRIPTION
**Pull request by elizaan**
### Does this PR close any open issues?
Closes #64 

### Give a longer description of what this PR addresses and why it's needed
previously, we used to calculate the trend change based on the RSD (relative standard deviation).
Now, we do fit our data in an exponential curve, and based on the exponent value, we decide on moderately flattens or drastically.

### **Covid data:**
  1. The UpSet plot:
  Screenshot 2024-09-09 at 5 49 39 PM

  2. The fitted curve:
  Screenshot 2024-09-09 at 5 58 57 PM

  3. The generated trend change text:
  
  _"The intersection sizes peak at a value of 281 and then moderately flatten down to 1"_

### **Tennis data:**
  1. The UpSet plot
  Screenshot 2024-09-09 at 5 52 09 PM

  2. The fitted curve:
  Screenshot 2024-09-09 at 6 00 01 PM

  3. The generated trend change text:
  
  _"The intersection sizes peak at a value of 23 and then moderately flatten down to 1"_

###  **Movies data:**
  1. The UpSet plot:
  Screenshot 2024-09-09 at 5 54 38 PM

  2. The fitted curve:
  Screenshot 2024-09-09 at 6 09 08 PM

  3. The generated trend change text:
  
  _"The intersection sizes peak at a value of 1191 and then drastically flatten down to 1"_

### Have you added or updated relevant tests?
- [ ] Yes
- [ ] No changes are needed

### Have you added or updated relevant documentation?
- [ ] Yes
- [ ] No changes are needed

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [ ] ...

### CODE REVIEW

File: src/alttxt/tokenmap.py [hunk 1]
```diff
@@ -676,23 +678,24 @@ def calculate_set_divergence(self):
...
-            return IntersectionTrend.GRADUAL.value
-        else:
+        popt, pcov = curve_fit(lambda x, a, b, c: a*np.exp(-b*x)+c, x, y, p0=[max(y), 0.1, min(y)])
+        a, b, c = popt   
+
+        if b > 0.5:
             return IntersectionTrend.DRASTIC.value
+        else:
+            return IntersectionTrend.MODERATE.value
+
```
Comment (JakeWags) [inline src/alttxt/tokenmap.py]: Is there no more option for GRADUAL?

**Review by JakeWags — APPROVED** [review]
Looks good! 👍

**Comment by JakeWags** [comment]
Any time you add dependencies make sure to add them to `requirements.txt`. I have done that for you in https://github.com/visdesignlab/upset-alt-txt-gen/pull/69/commits/2fa0bf7c55c66ec9b42d491f52a69d23f768b9bd

File: src/alttxt/tokenmap.py (hunk 1 above)
Comment (elizaan) [reply src/alttxt/tokenmap.py]: The new changes had to be matched with paper writing.
Screenshot 2024-09-10 at 9 27 10 AM

**Comment by elizaan** [comment]
> Any time you add dependencies make sure to add them to `requirements.txt`. I have done that for you in [2fa0bf7](https://github.com/visdesignlab/upset-alt-txt-gen/commit/2fa0bf7c55c66ec9b42d491f52a69d23f768b9bd)

Thank you! Will do from next on

**Comment by elizaan** [comment]
Added some changes in my latest code for linear fits:
### **Organizations Data**
1. UpSet:
Screenshot 2024-09-10 at 9 42 50 AM

2. Curve Fit:
Screenshot 2024-09-10 at 9 45 21 AM

3. In text description:
_The intersection sizes peak at a value of 9 and then moderately flatten down to 1_

**### Tennis Data**

1. UpSet:
Screenshot 2024-09-10 at 9 43 03 AM

2. Curve fit:
Screenshot 2024-09-10 at 9 45 30 AM

3. In text description:
_The intersection sizes peak at a value of 33 and then moderately flatten down to 23_

**Review by JakeWags — APPROVED** [review]
Looks great 😄

**Pull request merged**